mcp_server/gitlab_tools.py

```diff
@@ -135,6 +145,41 @@ async def get_internal_rhel_branches(
         raise ToolError(f"Failed to get branches for package {package}: {ex}")
 
 
+async def clone_repository(
+    repository: Annotated[str, Field(description="Repository URL to clone")],
+    clone_path: Annotated[str, Field(description="Absolute path where to clone the repository")],
+    branch: Annotated[str, Field(description="Branch to clone")] = None,
+) -> str:
+    """
+    Clones the specified repository to the given local path.
+    """
+    # Add authentication token to repository URL if available
+    authenticated_url = _get_authenticated_url(repository)
+
+    command = [
+        "git",
+        "clone",
+        "--single-branch",
+        "--branch",
+        branch,
+        authenticated_url,
+        clone_path,
+    ]
+
+    proc = await asyncio.create_subprocess_exec(command[0], *command[1:])
+    if await proc.wait():
+        raise ToolError(f"Failed to clone repository {repository}")
+
+    # Remove all remote references after cloning
```

Comment:
The comment says all but the command removes only `origin`. Sure, only `origin` should exist as it is autocreated, but the comment should reflect that. Also, to be 100% safe, we should really remove all remotes. And we can't rely on the name `origin` as it is possible to change that for example with a global git config.
I think we should do something like `for remote in $(git remote); do git remote remove $remote; done`. And it would probably be more elegant to use GitPython for this (and other git operations).